docs/_tutorials/migration_v2.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Migrating to V2
+order: 1
+slug: migration-v2
+lang: en
+layout: tutorial
+permalink: /tutorial/migration-v2
+---
+# Migrating to v2.x
+
+<div class="section-content">
+This guide will walk you through the process of updating your app from using `bolt@v1.x` to `bolt@2.x`. 
+</div> 
+
+---
+
+## Upgrading your listeners to `async`
+
+Listeners in your app should updated to `async` functions and `say()`,  `respond()`, and `ack()` should be prefaced with `await`.
+
+Before:
+
+```javascript
+app.action('some-action-id', ({action, ack, say}) => { 
+  ack();
+  say('hello world');
+})
+```
+
+After:
+
+```javascript
+app.action('some-action-id', async ({action, ack, say}) => { 
+  await ack();
+  await say('hello world');
+})
+```
+
+
+## Error Handling
+
+The recent changes in Bolt for JavaScript V2 have improved our ability to catch errors and filter them to the global error handler.  It is still recommended to manage errors in the listeners themselves instead of letting them propagate to the global handler when possible.
+
+### Handling Errors in Listeners with `try`/`catch`
+
+```javascript
+app.action('some-action-id', async ({action, ack, say, logger}) => { 
+  try {
+    await ack();
+    await say('hello world');
+  } catch (error) {
+    logger.error(error);
+    // handle error
+  }
+})
+```
+
+### Handling Errors with the Global Error Handler
+
+```javascript
+app.error((error) => {
+  // Check the details of the error to handle cases where you should retry sending a message or stop the app
+  console.error(error);
+});
+```
+
+Other error related changes include:
+
+- When your listener doesn’t call `ack` within the 3 section time limit, we log the failure instead of throwing an error.
+- If you have multiple errors at once when running middleware, Bolt for Javascript will return a wrapper error with a `code` parameter of `slack_bolt_multiple_listener_error` and an `original` parameter that contains an array of all of the errors. 
```

Comment:
We don't want people using the string value of error codes directly. It's much more future-proof if they use the enums. Also fix on `original` to `originals` (plural).

```suggestion
- If multiple errors occur when processing multiple listeners for a single event, Bolt for Javascript will return a wrapper error with a `code` property of `ErrorCode.MultipleListenerError` and an `originals` property that contains an array of the individual errors. 
```